src/_pytest/config/argparsing.py

```diff
@@ -127,19 +128,24 @@ def parse_known_and_unknown_args(
         args = [str(x) if isinstance(x, py.path.local) else x for x in args]
         return optparser.parse_known_args(args, namespace=namespace)
 
-    def addini(self, name, help, type=None, default=None):
+    def addini(self, name, help, type=None, default=None, choices=None):
         """ register an ini-file option.
 
         :name: name of the ini-variable
-        :type: type of the variable, can be ``pathlist``, ``args``, ``linelist``
-               or ``bool``.
+        :type: type of the variable, can be ``pathlist``, ``args``, ``linelist``,
+               ``bool``, or ``type``.
         :default: default value if no ini-file option exists but is queried.
 
         The value of ini-variables can be retrieved via a call to
         :py:func:`config.getini(name) <_pytest.config.Config.getini>`.
         """
-        assert type in (None, "pathlist", "args", "linelist", "bool")
-        self._inidict[name] = (help, type, default)
+        # TODO: callable type?
```

Comment:
Not sure we need to make this a callable, after all the user can always call the callable before passing the choices to this function... or do you have a specific use case in mind?

Reply:
It was meant to work like `type` in argparse then - which is more universal than having choices (only).

Reply:
Oh I thought you meant `choices` being able to be a callable, my bad.

We can add `type` here no problem, following the same semantics that `type` and `choices` have in `argparser`. I will leave this up to you if you prefer to do this in this PR or do it separately.